multiversx/mx-chain-go
# Added the possibility to define more protocol IDs for p2p networks

## Reasoning behind the pull request
- added the possibility to define more protocol IDs for p2p networks
  
related to https://github.com/multiversx/mx-chain-communication-go/pull/48

## Proposed changes
- 
- 
- 

## Testing procedure
- start a system testnet. Make sure that when we upgrade the nodes (epoch 1) the seednode is compiled from this branch and the p2p.toml file for the seeder is fetched from this branch. Make sure that after the upgrade process for all nodes their p2p.toml is fetched from this branch. The test should process blocks normally
- start a half-half test with this branch and `rc/v1.7.next` branch. The seeder must be compiled from this branch & its p2p.toml should be used from this branch. All nodes upgraded to this branch variant should use their p2p.toml fetched from this branch.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
